app/views/questionnaire.py

```diff
@@ -216,18 +211,20 @@ def get_timeout_continue(eq_id, form_type, collection_id):  # pylint: disable=un
     return 'true'
 
 
-@questionnaire_blueprint.route('session-expired', methods=["GET"])
+@questionnaire_blueprint.route('session-expired', methods=["POST"])
 @login_required
-def get_session_expired(eq_id, form_type, collection_id):  # pylint: disable=unused-argument
+def post_session_expired(eq_id, form_type, collection_id):  # pylint: disable=unused-argument
     session_storage.clear()
     logout_user()
-    theme = g.schema_json.get('theme', None)
-    logger.debug("theme selected", theme=theme)
-    return render_theme_template(theme, template_name='session-expired.html',
-                                 legal_basis=g.schema_json['legal_basis'],
+    return redirect(url_for('.get_session_expired', eq_id=eq_id, form_type=form_type, collection_id=collection_id))
```

Comment:
instead of redirecting can't you just `return self.get_session_expired(eq_id, form_type, collection_id)`
Given you are essentially redirecting to the same page?